ros2doctor/ros2doctor/api/__init__.py

```diff
@@ -100,6 +101,8 @@ def generate_reports(*, categories=None) -> List[Report]:
     """
     reports = []
     for report_entry_pt in iter_entry_points('ros2doctor.report'):
+        report = Report('')  # default to invalid
+        report_category = ''
```

Comment:
I'm slightly concerned that making `report_category` an empty string can lead to a difficult-to-find bug.  In particular, if `categories` happens to contain an empty string, and the try/except blocks below throw an exception, we could insert a bogus Report.  Instead, I'll suggest that you move the if/else block at [119](https://github.com/ros2/ros2cli/pull/332/files#diff-26940033a37b6de29cd2efa292e66629R119) into the try block directly above it.  That way, this will never be an issue.

Reply:
You moved the block below (great!) but you forgot to remove these initializations, which should no longer be necessary.

Reply:
oops 😳